src/tools/rest-backend/model_user.hpp

```diff
@@ -0,0 +1,298 @@
+	}
+	/**
+                 * Getter for the password hash.
+                 * @return Password hash as string
+                 */
+	std::string getPasswordHash () const
+	{
+		return this->get<std::string> ();
+	}
+
+	/**
+                 * Setter for the email
+                 * @param email Email as string
+                 */
+	void setEmail (std::string email)
+	{
+		kdb::Key k;
+		try
+		{
+			k = this->getSubkey (ELEKTRA_REST_MODEL_USER_META_EMAIL);
+		}
+		catch (kdbrest::exception::SubkeyNotFoundException & e)
+		{
+			k.setName (this->getName () + std::string ("/") + ELEKTRA_REST_MODEL_USER_META_EMAIL);
+		}
+		k.set<std::string> (email);
+		this->addSubkey (k);
+	}
+	/**
+                 * Getter for the email.
+                 * @return Email as string
+                 */
+	std::string getEmail ()
+	{
+		try
+		{
+			kdb::Key k = this->getSubkey (ELEKTRA_REST_MODEL_USER_META_EMAIL);
+			return k.get<std::string> ();
+		}
+		catch (kdbrest::exception::SubkeyNotFoundException & e)
+		{
+			return "";
+		}
+	}
+
+	/**
+                 * Setter for the rank.
+                 * 3 = Admin
+                 * 2 = Moderator
+                 * 1 = User
+                 * @param rank Rank as integer
+                 */
+	void setRank (int rank)
+	{
+		kdb::Key k;
+		try
+		{
+			k = this->getSubkey (ELEKTRA_REST_MODEL_USER_META_RANK);
+		}
+		catch (kdbrest::exception::SubkeyNotFoundException & e)
+		{
+			k.setName (this->getName () + std::string ("/") + ELEKTRA_REST_MODEL_USER_META_RANK);
+		}
+		k.set<int> (rank);
+		this->addSubkey (k);
+	}
+	/**
+                 * Getter for the rank.
+                 * 3 = Admin
+                 * 2 = Moderator
+                 * 1 = User
+                 * @return Rank as integer
+                 */
+	int getRank ()
+	{
+		try
+		{
+			kdb::Key k = this->getSubkey (ELEKTRA_REST_MODEL_USER_META_RANK);
+			return k.get<int> ();
+		}
+		catch (kdbrest::exception::SubkeyNotFoundException & e)
+		{
+			return 1; // default is user rank
+		}
+	}
+
+	/**
+                 * Setter for the creation date.
+                 * @param created_at Timestamp when the user has been created
+                 */
+	void setCreatedAt (const long created_at)
+	{
+		kdb::Key k;
+		try
+		{
+			k = this->getSubkey (ELEKTRA_REST_MODEL_USER_META_CREATEDAT);
+		}
+		catch (kdbrest::exception::SubkeyNotFoundException & e)
+		{
+			k.setName (this->getName () + std::string ("/") + ELEKTRA_REST_MODEL_USER_META_CREATEDAT);
+		}
+		k.set<long> (created_at);
+		this->addSubkey (k);
+	}
+	/**
+                 * Getter for the creation date.
+                 * @return Timestamp when the user has been created
+                 */
+	long getCreatedAt ()
+	{
+		try
+		{
+			kdb::Key k = this->getSubkey (ELEKTRA_REST_MODEL_USER_META_CREATEDAT);
+			return k.get<long> ();
+		}
+		catch (kdbrest::exception::SubkeyNotFoundException & e)
+		{
+			return 0;
+		}
+	}
+
+	static bool less_than_username (User & l, User & r)
```

Comment:
Where would you place it? To me it makes sense that it is near the compared model.